### DISCUSSION TITLE
User-defined subdomains + API routes

### DISCUSSION THREAD

**Discussion by kennethacohen** [vercel/next.js#14636] [Help]
I'm trying to understand if this project is feasible using NextJS + API routes (i.e. avoiding a custom express server). 

What I'm trying to achieve:

- main static site would be at `site.com`
- users would be able to create accounts to essentially run a blog; each account would generate a subdomain they would define. so one subdomain per user: `usera.site.com`, `userb.site.com` etc.
- going to `usera.site.com/posts` would show posts for only `usera`, meaning I would probably want to GET `usera.site.com/api/v1/posts`

My current understanding is that I would want to set up something like NGINX to reverse proxy these requests to the main application, but haven't seen a pattern for this anywhere. Also possible I just need to go with the custom server route, but would like to avoid that as I've enjoyed dev'ing with the built in Next APIs.

Would appreciate any help here ❤️

**Comment by lucasbastianik** [comment]
+1

**Comment by weyert** [comment]
Can't you use a catch-all DNS record which redirects all traffic to `*.site.com`? And then you check in `getServerProps` the `host`-header and then deal with it accordingly.
